extensions/amp-analytics/0.1/events.js

```diff
@@ -147,13 +148,45 @@ export class CustomEventTracker extends EventTracker {
     const targetReady =
         this.root.getElement(context, selector, selectionMethod);
 
-    let observers = this.observers_[eventType];
-    if (!observers) {
-      observers = new Observable();
-      this.observers_[eventType] = observers;
+    const isSandboxEvent = startsWith(eventType, 'sandbox-');
+
+    // Push recent events if any.
+    let buffer;
```

Comment:
it's not a good idea to do 
```const buffer = .....```
 because if buffer doesn't exist I'll have to create buffer, and `buffer[eventType] = []` won't work as `this.buffer_[eventType] = []`.
I did however try to share as much logic as I can. PTAL